### PR TITLE
Release preparations for 2.0

### PR DESCRIPTION
**Pull request by sgraf812**
I mostly completed the ChangeLog from looking at [this list](https://github.com/haskell/happy/commits/master?since=2021-10-17&until=2024-09-16).

If someone would give me the necessary hackage credentials (user sgraf812), I would be happy to do the hackage release.
I would just use `nix run nixpkgs#hkgr -- publish` to do so. Or is there some documented procedure one should follow for releases? At any rate, we should document the procedure. Perhaps in `CONTRIBUTING.rst`?

### CODE REVIEW

**Comment by Ericson2314** [comment]
https://github.com/haskell/happy/blob/652c4ca25df9a612e4df5044c5c69ae90f356925/doc/introduction.rst?plain=1#L147-L156 do you want to be an official maintainer? :)

**Review by Ericson2314 — APPROVED** [review]

**Comment by sgraf812** [comment]
> do you want to be an official maintainer? :)

I would be honored to, and added myself to the list :) 🎉

**Review by Ericson2314 — APPROVED** [review]
If @int-index or @andreasabel signs off on this, I'll give @sgraf812 the haddock perms :)

**Comment by int-index** [comment]
> If @int-index or @andreasabel signs off on this, I'll give @sgraf812 the haddock perms :)

Please do!

**Comment by sgraf812** [comment]
Thanks :) I'll wait for another 9 hours to give @andreasabel a chance to react before doing the release.

**Comment by andreasabel** [comment]
Thanks for all the good work, @sgraf812 !

I did local tests of `cabal build`, `cabal test`, `cabal install`, and using the new Happy for building Agda and other projects.

Could you update the release candidate on hackage (https://hackage.haskell.org/package/happy-2.0/candidate) before pressing the final button?  The current one is
> Uploaded	by [int_index](https://hackage.haskell.org/user/int_index) at 2024-02-03T14:08:09Z

**Comment by sgraf812** [comment]
Here's a package candidate for `happy-lib`: https://hackage.haskell.org/package/happy-lib-2.0/candidate

And the one for `happy`: https://hackage.haskell.org/package/happy-2.0/candidate

Shall I do `cabal gen-bounds` as suggested?

File: ChangeLog.md [hunk 1]
```diff
@@ -2,21 +2,35 @@
 
 ## 2.0
 
-The main focus of this release was modularizing Happy.
+There are two main breaking changes in this release:
 
-* Generated parsers now activate the language extension `-XNoStrictData` without
-  which every use of a happy parser would lead to an immediate crash (#273).
-  This causes us to drop support for GHC < 8.0.
+1. Removal non-array, non-GHC modes, so flags `-ag` are the default now and
```
Comment (andreasabel) [inline ChangeLog.md:7]: "Removal of" or "Remove"

**Comment by andreasabel** [comment]
Reading the changelog in another context (on Hackage), I found a typo.

> Here's a package candidate for `happy-lib`: [hackage.haskell.org/package/happy-lib-2.0/candidate](https://hackage.haskell.org/package/happy-lib-2.0/candidate)

Thanks!

Atm, you are the sole maintainer, this is probably not intended:

> Package maintainers
> [sgraf812](https://hackage.haskell.org/user/sgraf812)

> Shall I do `cabal gen-bounds` as suggested? 

Na, I would ignore this.  
Let's wait for real breakage, then one can still place upper bounds via Hackage revisions.

> Also the missing documentation seems strange (I had uploaded it as well)

The `happy-lib` does have the documentation.
Executables usually do not have any documentation, so this looks alright, if you ask me.
(In contrast, Agda is both an executable and a library, so it has documentation.)

File: ChangeLog.md (hunk 1 above)
Comment (sgraf812) [reply ChangeLog.md:7]: Thanks, fixed (`Remove`)!

**Comment by sgraf812** [comment]
> Atm, you are the sole maintainer, this is probably not intended:

Thanks for pointing me to the link where I could change it.

> The happy-lib does have the documentation.

That was due to an earlier experiment on my part where I simply made the main component expose all modules directly and deleted all sublibraries (apologies, should have reverted). The latter step was sadly necessary to see the full docs, which seems like a no go after the discussion in #288.

I just re-uploaded the candidate with the state where we have multiple private libraries and after fixing the ChangeLog.

I'm rather reluctant to get rid of the private libraries again (#288), so let's just release as-is and reupload docs when `cabal` has sorted out the situation wrt. merging documentation of multiple private sublibraries (https://github.com/haskell/cabal/issues/10368). I honestly do not expect much use of `happy-lib` anyway, and the few users that we attract are likely experts.

A quick glance at https://hackage.haskell.org/package/happy-lib-2.0/candidate and a nod would be appreciated.

File: ChangeLog.md (hunk 1 above)
Comment (andreasabel) [reply ChangeLog.md:7]: So the candidates still have the typo, weirdly...

**Review by andreasabel — COMMENTED** [review]
LGTM ! (except that the typo still seems to plague the candidates).

File: ChangeLog.md (hunk 1 above)
Comment (sgraf812) [reply ChangeLog.md:7]: Perhaps that was a caching issue. Regardless, in https://hackage.haskell.org/package/happy-lib-2.0/changelog it does say `Removed`, not `Removal`, for me.

**Pull request merged**